ICU-23148 Migrate ICU4J publishing from Legacy OSSRH to Central Portal (#3534)
unicode-org/icu: docs/processes/release/tasks/release-build.md

## Building ICU4J Release Files

<span style="color:red">*Note: This instruction is reviewed for ICU4J 78.1. Some
description may not apply to previous ICU releases. If you need instructions on

> how to release older versions see the history of this file.*</span>

Reviewer: I don't think that Markdown symbols are honored inside HTML tags. So remove the pair of asterisks and replace with the appropriate `i` or `b` tag or update the `span` style attribute, if desired.

Author: Done

> The releasing done 100% from a GitHub Action.

Reviewer: ```suggestion
The releasing is done entirely via a GitHub Action.
```

## Prerequisites

None.

> The GitHub Action uses the proper JDK versioin, uses secrets to store the

Reviewer: ```suggestion
The GitHub Action uses the proper JDK version, uses secrets to store the
```

> gpg user and password, Sonatype token, etc.

Reviewer: ```suggestion
GPG user and password, Sonatype token, etc.
```

## Building and publishing the ICU4J release files

A final (official) release files will be posted to the Maven Central Repository
as well as the ICU GitHub releases.

This is all done by a GitHub Action,
see [publish/release.md](publish/release.md),
the _"Release - ICU4J publish to Maven Central"_ section.

The previous step stages the Maven artifacts to Sonatype, but does
not automatically push them to Maven Central. \

> It can be enabled, but we chose no to, so that a human can do a last sanity check. \

Reviewer: ```suggestion
It can be enabled, but we chose no to enable it by default so that a human can do a last sanity check. \
```

So someone must login to Sonatype, check that everything looks fine, and approve. \
To do that log on to the [Sonatype Central Portal -- Namespaces](https://central.sonatype.com/publishing/namespaces). \
Select the "Deployments" tab. \
Check the files staged there, compare to a previous public release in Maven Central
(for example [ICU4J 77.1](https://repo1.maven.org/maven2/com/ibm/icu/icu4j/77.1/)
and [ICU4J Charset 77.1](https://repo1.maven.org/maven2/com/ibm/icu/icu4j-charset/77.1/)) \

> Once you confirm the contents, approve it. \

Reviewer: If these are steps for doing an approval of a deployment, then put them in an ordered list.

If the approval step is something that has a technical step to it (clicking a putting, updating a setting in some UI), then describe that with words. The screenshot might help, but words are still important.

**TODO: update this doc with screenshot(s) once we have a release staged.**

<span style="color:red">**Note:** only someone with a Sonatype account that
was authorized for `com.ibm.icu` can approve. \

> Currently these are the ICU Robot, Elango, Mihai and Yoshito.</span>

Reviewer: Remove this line referring to people by name. You can mention that the current list of people with access is stored in the team shared folder.
